configs/module.go

```diff
@@ -200,15 +205,15 @@ func (m *Module) appendFile(file *File) hcl.Diagnostics {
 
 	for _, pm := range file.ProviderMetas {
 		// TODO(paddy): pm.Provider is a string, but we need to build an addrs.Provider out of it somehow
-		if existing, exists := m.ProviderMetas[addrs.NewLegacyProvider(pm.Provider)]; exists {
+		if existing, exists := m.ProviderMetas[addrs.NewDefaultProvider(pm.Provider)]; exists {
```

Comment:
The comment above this makes me think that this ought to be a lookup in the local-to-FQN mapping table, rather than just `NewDefaultProvider`. I assume this is dealing with the labels for blocks like this:

```hcl
terraform {
  provider_meta "google" {
    # ...
  }
}
```

...in which case, I think in the new model of things the intent of this would be for the metadata to be attached to whatever provider has the local name `google`.

What do you think? :thinking:

(I raise this now only because I worry that once this isn't calling `NewLegacyProvider` anymore it will be harder to find it to iterate on it further.)

Reply:
TBH that got caught up in my "get tests passing" find & replace mess and I haven't _actually_ looked at the `ProviderMetas` yet - it's a pending TODO.